torch/_inductor/index_propagation.py

```diff
@@ -54,7 +54,9 @@ def identity(value: Any) -> Any:
 
     @staticmethod
     def constant(value: Union[int, float, bool], dtype: torch.dtype) -> TypedExpr:
-        if is_boolean_dtype(dtype):
+        if isinstance(value, sympy.Expr):
+            expr = value
```

Comment:
`ops.constant` shouldn't be called with `sympy.Expr`. Instead it should be `ops.index_expr`.

Reply:
So it turns out that it's easy enough to change the ops.constant call in the lowering to ops.index_expr, but now I am wondering, is there really any material difference between these two calls? Like, it's not like you're going to get worse code if you call index_expr with a constant, no?

Reply:
Ideally I think they should be similar but in practice they are materially different. `ops.constant` will respect the dtype argument and generate code like `tl.full([1, 1], 256, tl.int32)` whereas `ops.index_expr` goes through the sympy expression printer which gives an integer literal `256` which ignores the specified dtype.

Reply:
in any case I fixed it, plz review lol